insights/parsers/nginx_conf.py

```diff
@@ -0,0 +1,338 @@
+            elif spacey(entry):
+                # don't delete comments
+                if "#" not in self[:i]:
+                    list.__delitem__(self, i)
+
+
+def _handle_key_value(t_dict, key, value):
+    """
+    Function to handle key has multi value, and return the values as list.
+    """
+    if key in t_dict:
+        val = t_dict[key]
+        if isinstance(val, dict):
+            val = [val]
+        val.append(value)
+        return val
+    return value
+
+
+def depth(l):
+    """
+    Function to count the depth of a list
+    """
+    if isinstance(l, list):
+        return 1 + max(depth(item) for item in l)
+    else:
+        return 0
+
+
+def convert_nginx_list_to_dict(li):
+    """
+    After parsed by create_parser(), the result is like following, it is better to convert to dict for convenience.
+    Example: [['user', 'root'], ['worker_processes', '5'], ['error_log', 'logs/error.log'], ['pid',
+    'logs/nginx.pid'], ['worker_rlimit_nofile', '8192'], [['events'], [['worker_connections', '4096']]], [['mail'],
+    [['server_name', 'mail.example.com'], ['auth_http', 'localhost:9000/cgi-bin/auth'], [['server'], [['listen',
+    '143'], ['protocol', 'imap']]]]], [['http'], [['include', '/etc/nginx/conf.d/*.conf'], ['index', 'index.html
+    index.htm index.php'], ['log_format', 'main  \'$remote_addr - $remote_user [$time_local] "$request" \'\n\'$status
+    $body_bytes_sent "$http_referer" \'\n\'"$http_user_agent" "$http_x_forwarded_for"\''], ['default_type',
+    'application/octet-stream'], ['access_log', 'logs/access.log  main'], ['sendfile', 'on'], ['tcp_nopush', 'on'],
+    ['server_names_hash_bucket_size', '128'], [['server'], [['listen', '80'], ['server_name', 'domain1.com
+    www.domain1.com'], ['access_log', 'logs/domain1.access.log  main'], ['root', 'html'], [['location', '~',
+    '\\.php$'], [['fastcgi_pass', '127.0.0.1:1025']]]]], [['server'], [['listen', '80'], ['server_name', 'domain2.com
+    www.domain2.com'], ['access_log', 'logs/domain2.access.log  main'], [['location', '~', '^/(images|javascript|
+    js|css|flash|media|static)/'], [['root', '/var/www/virtual/big.server.com/htdocs'], ['expires', '30d']]],
+    [['location', '/'], [['proxy_pass', 'http://127.0.0.1:8080']]]]], [['map', '$http_upgrade',
+    '$connection_upgrade'], [['default', 'upgrade'], ["''", 'close']]], [['upstream', 'websocket'], [['server',
+    '10.66.208.205:8010']]], [['upstream', 'big_server_com'], [['server', '127.0.0.3:8000 weight=5'], ['server',
+    '127.0.0.3:8001 weight=5'], ['server', '192.168.0.1:8000'], ['server', '192.168.0.1:8001']]], [['server'],
+    [['listen', '80'], ['server_name', 'big.server.com'], ['access_log', 'logs/big.server.access.log main'],
+    [['location', '/'], [['proxy_pass', 'http://big_server_com']]]]]]]]
+
+    After converting, it should be like following:
+
+    For the section, the section name would be a dict key, and content embraced by brace would be value. In some
```

Comment:
I think an actual example of the conversion of (some of) the above into dict format would be good here.